### PR TITLE
compat: sqlalchemy deprecations

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -77,6 +78,16 @@ def _is_sqlalchemy_connectable(con):
         return False
 
 
+def _gt14() -> bool:
```
Comment (jreback) [inline pandas/io/sql.py:81]: i think we have a test decorator for this type of thing right?

**Review by jreback — COMMENTED** [review]
looks fine but comment on the way u r getting the version

File: pandas/io/sql.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/sql.py:81]: we have a decorator to skip a test depending on the version, not to re-write the test

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok i think we should prob backport this any objections?

**Comment by jreback** [comment]
cc @simonjayhawkins its possible may need to backport things like this for 1.2.4

**Comment by jorisvandenbossche** [comment]
Yeah, I think this might be useful to backport

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.2.x

**Pull request merged**